### PR TITLE
db: exposes threadid from js-threads for use in db creation

### PR DESCRIPTION
**Pull request by andrewxhill**
Signed-off-by: andrew <andrew@textile.io>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,16 +58,18 @@
   "dependencies": {
     "@improbable-eng/grpc-web": "^0.12.0",
     "@textile/threads-client-grpc": "^0.1.12",
+    "@textile/threads-core": "0.0.3-alpha.1",
     "@types/google-protobuf": "^3.7.2",
+    "@types/multiaddr": "^7.2.0",
```
Comment (carsonfarmer) [inline package.json]: I just pushed another release that means you don't have to explicitly add these deps. But ok to leave them here for now.

File: src/__test__/index.test.ts [hunk 2]
```diff
@@ -60,8 +57,8 @@ const createPerson = (): Person => {
 }
 
 describe('Client', function() {
-  const bytes = Buffer.concat([Buffer.from(encode(0x01)), Buffer.from(encode(0x55)), randomBytes(32)])
-  const dbID = multibase.encode('base32', bytes).toString()
+  const threadId = ThreadID.fromRandom()
+  const dbID = threadId.stringOfBase('base32')
```
Comment (carsonfarmer) [inline src/__test__/index.test.ts]: Ah I see. The method you wanted here was `string()` (which does default to base32). But I've created an issue to actually switch to `toString()` for all this to simplify things.

File: src/index.ts [hunk 3]
```diff
@@ -28,13 +28,15 @@ import {
...
 } from '@textile/threads-client-grpc/api_pb'
+import { ThreadID } from '@textile/threads-core'
 import { encode, decode } from 'bs58'
 import * as pack from '../package.json'
 import { ReadTransaction } from './ReadTransaction'
 import { WriteTransaction } from './WriteTransaction'
 import { Config, BaseConfig } from './config'
 import { JSONQuery, Instance, InstanceList } from './models'
 
+export { ThreadID }
```
Comment (carsonfarmer) [inline src/index.ts]: 👍

File: package.json (hunk 1 above)
Comment (andrewxhill) [reply package.json]: udpated

**Comment by andrewxhill** [comment]
these tests will fail until we release the next go-threads version i believe. going to move ahead

File: src/__test__/index.test.ts (hunk 2 above)
Comment (andrewxhill) [reply src/__test__/index.test.ts]: ah okay. same outcome. i'll change it with future update

**Pull request merged**